diff --git a/libevhtp/CMakeLists.txt b/third_party/libevhtp//libevhtp/CMakeLists.txt
index 481ddd0..02b55d8 100644
--- a/libevhtp/CMakeLists.txt
+++ b/third_party/libevhtp//libevhtp/CMakeLists.txt
@@ -17,6 +17,8 @@ include(TestBigEndian)
 
 check_function_exists(strndup HAVE_STRNDUP)
 check_function_exists(strnlen HAVE_STRNLEN)
+check_function_exists(strcasecmp, HAVE_STRCASECMP)
+check_function_exists(strncasecmp, HAVE_STRNCASECMP)
 
 check_include_files(stdlib.h    HAVE_STDLIB_H)
 check_include_files(string.h    HAVE_STRING_H)
@@ -35,6 +37,8 @@ check_include_files(sys/tree.h  HAVE_SYS_TREE)
 check_include_files(sys/queue.h HAVE_SYS_QUEUE)
 check_include_files(sys/un.h    HAVE_SYS_UN)
 check_include_files(sys/types.h HAVE_SYS_TYPES_H)
+check_include_files(arpa/inet.h HAVE_ARPA_INET_H)
+check_include_files(compat/sys/queue.h.in HAVE_LOCAL_SYS_QUEUE)
 
 check_type_size("int" SIZEOF_INT)
 check_type_size("long" SIZEOF_LONG)
@@ -54,10 +58,10 @@ set(LIBEVHTP_SOURCE_FILES
     parser.c
     log.c)
 
-find_package(LibEvent REQUIRED)
+find_package(Libevent REQUIRED)
 list(APPEND LIBEVHTP_EXTERNAL_LIBS ${LIBEVENT_LIBRARIES})
 list(APPEND LIBEVHTP_EXTERNAL_INCLUDES ${LIBEVENT_INCLUDE_DIRS})
-list(APPEND package_deps LibEvent)
+list(APPEND package_deps Libevent)
 
 set(evhtp_dir_headers
   "include/evhtp/evhtp.h"
@@ -75,12 +79,23 @@ if(NOT EVHTP_DISABLE_SSL)
 endif()
 
 if(NOT EVHTP_DISABLE_EVTHR)
-    find_package(Threads)
-    if(TARGET Threads::Threads)
-        list(APPEND LIBEVHTP_SOURCE_FILES  thread.c)
-        list(APPEND LIBEVHTP_EXTERNAL_LIBS Threads::Threads)
-        list(APPEND package_deps Threads)
-        list(APPEND evhtp_dir_headers "include/evhtp/thread.h")
+    if (WIN32)
+        # pthreads for windows vcpkg seems not to have proper cmake integration
+        set(VCPKG_INSTALL_DIR "C:/vcpkg/installed/x64-windows/")
+        find_path(PTHREADS_INC NAMES pthread.h PATHS "${VCPKG_INSTALL_DIR}/include/" REQUIRED)
+        find_library(PTHREADS_LIBS NAMES pthreadVC3 PATHS "${VCPKG_INSTALL_DIR}/lib/" REQUIRED)
+            list(APPEND LIBEVHTP_SOURCE_FILES  thread.c)
+            list(APPEND LIBEVHTP_EXTERNAL_LIBS ${PTHREADS_LIBS})
+            list(APPEND LIBEVHTP_EXTERNAL_INCLUDES ${PTHREADS_INC})
+            list(APPEND evhtp_dir_headers "include/evhtp/thread.h")
+    else()
+        find_package(Threads)
+        if(TARGET Threads::Threads)
+            list(APPEND LIBEVHTP_SOURCE_FILES  thread.c)
+            list(APPEND LIBEVHTP_EXTERNAL_LIBS Threads::Threads)
+            list(APPEND package_deps Threads)
+            list(APPEND evhtp_dir_headers "include/evhtp/thread.h")
+        endif()
     endif()
 endif()
 
