diff --git a/CMakeLists.txt b/CMakeLists.txt
index dfb441e..1f7f384 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -173,7 +173,7 @@ target_link_libraries(azure-storage-lite Threads::Threads ${CURL_LIBRARIES} ${EX
 if(MSVC)
   target_compile_options(azure-storage-lite PRIVATE /W4 /WX /MP)
 elseif(CMAKE_CXX_COMPILER_ID MATCHES "Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
-  target_compile_options(azure-storage-lite PRIVATE -Wall -Wextra -Werror -pedantic)
+  target_compile_options(azure-storage-lite PRIVATE -Wall -Wextra -Werror -pedantic -Wno-deprecated)
 endif()
 
 if(BUILD_ADLS)
